### ISSUE TITLE
12$: TypeScript: error on non-constant computed property names in declare classes and interfaces

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#943]
quick-lint-js does not complain about the below issues, but it should:
```typescript
interface I {
    ['hel' + 'lo'](): any;  // Invalid TypeScript.
}
declare class C {
    ['hel' + 'lo'](): any;  // Invalid TypeScript.
}
```
